### PR TITLE
Format Rust in precommit and lint+format scripts

### PR DESCRIPTION
**Pull request by wbinnssmith**
Depends on #6200.

This:
* Runs `rustfmt` on staged files in a precommit hook just as we do JS with prettier. 
* Uses `cargo fmt` in `lint` and `format` yarn scripts.
* Formats existing code.

Notes:
* Also updates husky to version 6 using typicode/husky-4-to-6, which uses files in a directory instead of package.json config.
* Set the `tab_spaces` configuration to 2 spaces to reduce churn with existing code. The Rust community and rustfmt seems to have settled on 4 spaces a default. Happy to discuss and change.

Test Plan: 
* Use rust-analyzer VSCode extension with automatic formatting and verify no diff churn on save.
* Make a change to a Rust file with unnecessary formatting and save without formatting the file. Commit the file and verify the unnecessary formatting is removed.
* Make a change to a Rust file with unnecessary formatting and run `yarn lint` and verify it fails. Run `yarn format` and verify the formatting is removed.

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 3.21s | -14.00ms |
| Cached | 457.00ms | +4.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 352.00ms | -26.00ms 🚀 |
| dist/legacy/index.76ce4b0c.js | 1.12kb | +0.00b | 1.07s | +400.00ms ⚠️ |
| dist/modern/index.b43fcc94.js | 1.12kb | +0.00b | 1.04s | +554.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 58.00ms | -3.00ms 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 77.00ms | -6.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 18.57s | -164.00ms |
| Cached | 710.00ms | -4.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 66.00ms | -21.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 60.00ms | +4.00ms ⚠️ |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.08m | -3.73s |
| Cached | 3.13s | +62.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/popup.119ff481.js | 171.12kb | +0.00b | 16.76s | -848.00ms 🚀 |
| dist/media-viewer.f15e56f4.js | 4.01kb | +0.00b | 10.43s | -1.50s 🚀 |
| dist/dropzone.d1e8004d.js | 3.46kb | +0.00b | 11.34s | -643.00ms 🚀 |
| dist/ResourcedEmojiComponent.5046c0ff.js | 2.19kb | +0.00b | 9.79s | +1.97s ⚠️ |
| dist/card.2888338b.js | 2.15kb | +0.00b | 11.04s | -606.00ms 🚀 |
| dist/images.ece03ae5.js | 1.91kb | +0.00b | 4.25s | -270.00ms 🚀 |
| dist/browser.8883c452.js | 1.85kb | +0.00b | 16.52s | -859.00ms 🚀 |
| dist/workerHasher.d607e0b8.js | 1.77kb | +0.00b | 16.66s | -854.00ms 🚀 |
| dist/heading6.5a6082c8.js | 1.53kb | +0.00b | 4.65s | -460.00ms 🚀 |
| dist/emoji.b5624cf5.js | 1.46kb | +0.00b | 4.25s | -270.00ms 🚀 |
| dist/heading5.88bc6249.js | 1.41kb | +0.00b | 4.65s | -460.00ms 🚀 |
| dist/16.fa5f411c.js | 1.36kb | +0.00b | 4.36s | +433.00ms ⚠️ |
| dist/16.d10db5a3.js | 1.32kb | +0.00b | 4.36s | +433.00ms ⚠️ |
| dist/heading4.a2524c70.js | 1.29kb | +0.00b | 4.65s | -460.00ms 🚀 |
| dist/layout.81f2196c.js | 1.28kb | +0.00b | 4.82s | +551.00ms ⚠️ |
| dist/divider.ed7ccd2e.js | 1.26kb | +0.00b | 4.25s | -270.00ms 🚀 |
| dist/quote.b6d8bffb.js | 1.25kb | +0.00b | 4.53s | -300.00ms 🚀 |
| dist/heading1.cc82b077.js | 1.19kb | +0.00b | 5.13s | +604.00ms ⚠️ |
| dist/16.90358f40.js | 1.18kb | +0.00b | 4.46s | +535.00ms ⚠️ |
| dist/panel-success.d5752da0.js | 1.06kb | +0.00b | 5.13s | +738.00ms ⚠️ |
| dist/panel-note.6a71a1f2.js | 1.05kb | +0.00b | 5.13s | +738.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.83da5db2.js | 1015.00b | +0.00b | 11.01s | -638.00ms 🚀 |
| dist/index.html | 119.00b | +0.00b | 1.44s | -241.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/card.df70c9b7.js | 54.67kb | +0.00b | 189.00ms | +12.00ms ⚠️ |
| dist/component.f25dcadb.js | 22.80kb | +0.00b | 87.00ms | -7.00ms 🚀 |
| dist/DatePicker.faca45d0.js | 21.38kb | +0.00b | 180.00ms | +17.00ms ⚠️ |
| dist/js.8167e2d5.js | 16.55kb | +0.00b | 87.00ms | -7.00ms 🚀 |
| dist/dropzone.3ae9a030.js | 16.15kb | +0.00b | 218.00ms | +16.00ms ⚠️ |
| dist/ui.15dd7acd.js | 14.30kb | +0.00b | 189.00ms | +12.00ms ⚠️ |
| dist/workerHasher.bc6346e1.js | 11.90kb | +0.00b | 166.00ms | -22.00ms 🚀 |
| dist/component.677fad45.js | 6.23kb | +0.00b | 87.00ms | -10.00ms 🚀 |
| dist/EmojiPickerComponent.4991ddd4.js | 3.68kb | +0.00b | 166.00ms | -11.00ms 🚀 |
| dist/png-chunks-extract.608f438d.js | 3.56kb | +0.00b | 87.00ms | -7.00ms 🚀 |
| dist/clipboard.6a6e5fbd.js | 3.00kb | +0.00b | 219.00ms | +17.00ms ⚠️ |
| dist/ResourcedEmojiComponent.5046c0ff.js | 2.19kb | +0.00b | 166.00ms | -11.00ms 🚀 |
| dist/card.2888338b.js | 2.15kb | +0.00b | 190.00ms | -27.00ms 🚀 |
| dist/feedback.facaac2e.js | 1.87kb | +0.00b | 180.00ms | +17.00ms ⚠️ |
| dist/16.89d45d15.js | 1.80kb | +0.00b | 127.00ms | +15.00ms ⚠️ |
| dist/workerHasher.31326e89.js | 1.77kb | +0.00b | 166.00ms | -22.00ms 🚀 |
| dist/16.00f25203.js | 1.47kb | +0.00b | 127.00ms | +15.00ms ⚠️ |
| dist/16.02056ae6.js | 1.47kb | +0.00b | 127.00ms | +12.00ms ⚠️ |
| dist/expand.4a483eb0.js | 1.39kb | +0.00b | 180.00ms | +17.00ms ⚠️ |
| dist/Modal.44e980e0.js | 1.29kb | +0.00b | 87.00ms | -7.00ms 🚀 |
| dist/16.015dfb9d.js | 1.26kb | +0.00b | 127.00ms | +15.00ms ⚠️ |
| dist/16.ec84a250.js | 1.26kb | +0.00b | 127.00ms | +15.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.9a500f44.js | 1019.00b | +0.00b | 189.00ms | +12.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.83da5db2.js | 1015.00b | +0.00b | 190.00ms | -27.00ms 🚀 |
| dist/simpleHasher.79137cb4.js | 767.00b | +0.00b | 166.00ms | -22.00ms 🚀 |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 15.61s | -179.00ms |
| Cached | 634.00ms | -3.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/Three.js | 580.60kb | +0.00b | 80.00ms | +4.00ms |

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/297153754762838536)**

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
```
Comment (mischnic) [inline .husky/pre-commit:3]: Was this file generated by that husky CLI?

**Comment by mischnic** [comment]
> Set the tab_spaces configuration to 2 spaces to reduce churn with existing code. The Rust community and rustfmt seems to have settled on 4 spaces a default. Happy to discuss and change.

I find that Rust is a bit harder to read with only 2 spaces (like determining which lines are inside the `if` body), for example here
Bildschirmfoto 2021-04-28 um 21 10 58

But we are using 2 spaces for JS as well, so 🤷‍♂️

File: .husky/pre-commit (hunk 1 above)
Comment (wbinnssmith) [reply .husky/pre-commit:3]: Yeah, I followed up a note in the file to clarify that.

**Review by devongovett — APPROVED** [review]

**Pull request merged**